go/kbfs/libkbfs/folder_block_manager.go

```diff
@@ -854,17 +855,19 @@ func getUnrefPointersFromMD(
 			// Can be zeroPtr in weird failed sync scenarios.
 			// See syncInfo.replaceRemovedBlock for an example
 			// of how this can happen.
-			if ptr != zeroPtr {
+			if ptr != zeroPtr && !ptrMap[ptr] {
 				ptrs = append(ptrs, ptr)
+				ptrMap[ptr] = true
 			}
 		}
 		for _, update := range op.allUpdates() {
 			// It's legal for there to be an "update" between
 			// two identical pointers (usually because of
 			// conflict resolution), so ignore that for quota
 			// reclamation purposes.
-			if update.Ref != update.Unref {
+			if update.Ref != update.Unref && !ptrMap[update.Unref] {
 				ptrs = append(ptrs, update.Unref)
+				ptrMap[update.Unref] = true
```

Comment:
I think it'd be better to use the map alone for accounting within this function, then producing the slice one time at the end. That would help prevent bugs where we forget to do both in each spot.